com.unity.render-pipelines.core/Runtime/Common/CollectionExtensions.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace UnityEngine.Rendering
+{
+    /// <summary>
+    /// A set of extension methods for collections
+    /// </summary>
+    public static class CollectionExtensions
+    {
+        /// <summary>
+        /// Removes a given count from the back of the given list
+        /// </summary>
+        /// <param name="inputList">The list to be removed</param>
+        /// <param name="elementsToRemoveFromBack">The number of elements to be removed from the back</param>
+        public static void RemoveBack<T>([NotNull] this IList<T> inputList, int elementsToRemoveFromBack)
```

Comment:
Use a generic over the list and then use it to change execution paths.
During compilation, the if branch will be stripped appropriately:

```csharp
public static bool TryRemoveExactRange<TList, TValue>([DisallowNull] TList list, uint startIndex, uint count)
    where TList: IList<TValue>
{
     // This branching is inlined at compilation
     if (typeof(TList) == typeof(List<TValue>))
     {
         var castedList __refvalue(__makeref(list), List<TValue>);
         castedList.RemoveRange(startIndex, count);
     }
     else
     {
         // IList implementation
     }
}
```

Reply:
If I do it like this, when I want to use it I need to specify the types, I've added another method as well.